### PR TITLE
Add dependency details

### PR DESCRIPTION
**Pull request by Glavin001**
Looking for style and content feedback! The current code definitely has room for improvement. The priority is what we want to show and how we want to show it, then I will go back and clean up the code afterwards.

---

See https://github.com/Unibeautify/website/issues/31 

Check out these pages:
- https://deploy-preview-76--unibeautify.netlify.com/docs/beautifier-prettier.html
- https://deploy-preview-76--unibeautify.netlify.com/docs/beautifier-php-cs-fixer.html
- https://deploy-preview-76--unibeautify.netlify.com/docs/executable-php-cs-fixer-php-cs-fixer.html

---

TODO:
- [x] https://docusaurus.io/docs/en/1.0.11/navigation.html#secondary-on-page-navigation

### CODE REVIEW

**Comment by Glavin001** [comment]
Screenshots of what it currently looks like:

image

image

image

File: scripts/generate-docs/BeautifierDoc.ts [hunk 1]
```diff
@@ -81,46 +86,118 @@ export default class BeautifierDoc extends Doc {
...
+    );
+    builder.append(
+      `**Note**: The \`LANGUAGE_NAME\` should be replaced with your desired supported language name, such as ${this.languages
+        .slice(0, 3)
+        .map(lang => `\`${lang.name}\``)
+        .join(", ")}, etc.`,
+    );
+
+    if (canResolveConfig) {
+      builder.header("Prefer Beautifier Specific Configuration File", 3);
```
Comment (szeck87) [inline scripts/generate-docs/BeautifierDoc.ts]: Put some blank space above this h3.  Or make the h3's stand out a bit more.  This particular one is hard to see that it's actually a heading.

**Comment by szeck87** [comment]
Having a separate page for the each of executables feels unnecessary.  I think it would be better to have one generic page that explains how to get the path and fix paths (if it's necessary to get the path based on your installation, see below).  And not too many people are going to watch two videos that amount to 15 minutes.

One other big thing missing is how to install the executables themselves.  I would personally just link to their website and if they don't have one, the GitHub repo.  As a Mac user I would find https://github.com/FriendsOfPHP/PHP-CS-Fixer#globally-homebrew and install it that way.  And doing that, I don't even need to specify the path in the Unibeautify config.  So I would also mention to have users try running `{exe command} --help` or something and if it works, then they don't need to bother with setting up paths in Unibeautify.

Most issues with Atom Beautify are with users not setting the path in the correct place (Atom Beautify settings -> Executables -> {Executable Name} -> Path).

**Comment by Glavin001** [comment]
Looking much better with the secondary navigation menu!

image

Awesome find @szeck87 ! Way to keep a close eye on those docs.

**Comment by Glavin001** [comment]
Adding Unibeautify version badge:

image

**Comment by szeck87** [comment]
I would describe all of the beautifier options before the example here https://deploy-preview-76--unibeautify.netlify.com/docs/options-for-beautifiers.html.  Also add details for the "Path" option in addition to "Prefer Beautifier Configuration".  Maybe the contents/general idea of https://deploy-preview-76--unibeautify.netlify.com/docs/executable-php-cs-fixer-php-cs-fixer.html could be moved there.  And https://deploy-preview-76--unibeautify.netlify.com/docs/executable-php-cs-fixer-php-cs-fixer.html could be home to how to install PHP-CS-Fixer.  So the [prereqs](https://deploy-preview-76--unibeautify.netlify.com/docs/beautifier-php-cs-fixer.html#prerequisites) would be install PHP-CS-Fixer itself, then add the path to the config file/editor settings.

For beautifiers with a node dependency, mention that they don't have to install (eslint is the example here) `unibeautify eslint @unibeautify/beautifier-eslint` if they are using an editor package/extension.  Those will already be included unless they prefer to use a global installation of `eslint`.  https://deploy-preview-76--unibeautify.netlify.com/docs/beautifier-eslint.html#install

**Comment by Glavin001** [comment]
Really love all of the feedback above! I'll need to re-read and go through the examples again later tonight / this weekend.

> Also add details for the "Path" option in addition to "Prefer Beautifier Configuration". 

Problem is the `path` option is for Executable options. And not every beautifier has Executables.
I could say something like "if your beautifier has executables listed under "Prerequisites" then read this" 😛 .

My goal was to automatically generate a much relevant information as possible. So by going to "PHP-CS-Fixer beautifier" page a user gets everything they need. I've since split it up into some more generic pages. I think this is something which may take a few iterations before we are happy with it. 

Thanks for working with me on this!

**Comment by szeck87** [comment]
> Problem is the path option is for Executable options. And not every beautifier has Executables.

Not everything will have prefer beautifier config either.  Just looking at the docs through the eyes of someone looking at them for the first time, it still seems daunting.

**Comment by szeck87** [comment]
@Glavin001 any more changes coming for this one?

**Comment by Glavin001** [comment]
@szeck87 : I need to review [your latest comments above](https://github.com/Unibeautify/website/pull/76#issuecomment-385063655) and make some more changes.

**Comment by Glavin001** [comment]
My latest beautifier and executable docs pages:

# Beautifier

image

# Executable 

image

**Comment by Glavin001** [comment]
OK I need to make a PR for Unibeautify and beautifier-php-cs-fixer next. Once those are merged then I'll update this website PR.

@szeck87 Any more improvements you see I could make in the screenshots above? 😃

**Comment by szeck87** [comment]
What needs updating in Unibeautify and php-cs-fixer?

**Comment by Glavin001** [comment]
> What needs updating in Unibeautify and php-cs-fixer?

Adding additional metadata fields for beautifier and dependency, such as `homepageUrl`, `installationUrl`, and `bugsUrl`, which are used to generate the website.

Will push in the next couple days.

**Comment by Glavin001** [comment]
@szeck87 here's the Unibeautify PR: https://github.com/Unibeautify/unibeautify/pull/100

Once Unibeautify PR is merged, I'll push out the PHP-CS-Fixer Beautifier PR -- need some more time.

**Comment by szeck87** [comment]
We can also use the published version of Docusaurus again, 1.0.15 fixed the custom_edit_url issue.

**Comment by Glavin001** [comment]
@szeck87 : Awesome to hear! I updated Docusaurus to v1.0.15 to fix #77.

**Comment by Glavin001** [comment]
PHP-CS-Fixer PR: https://github.com/Unibeautify/beautifier-php-cs-fixer/pull/17

**Comment by Glavin001** [comment]
@szeck87 : I'm done and ready for your review 😃 .

File: docs/config-file.md [hunk 2]
```diff
@@ -10,16 +10,25 @@ This means you can configure Unibeautify via:
 - A `unibeautify.config.js` file that exports an object.
 - A `"unibeautify"` key in your `package.json` file.
 
-The [options](options.md) to the configuration file are grouped by the respective language.
+The [options](options-for-languages.md) to the configuration file are grouped by the respective language.
```
Comment (Glavin001) [inline docs/config-file.md]: @szeck87 what're your thoughts on this rename? Do you think the language options should be shortened `options` and then beautifier's options be `options-for-beautifier`?

File: scripts/generate-docs/BeautifierDoc.ts [hunk 3]
```diff
@@ -35,24 +55,45 @@ export default class BeautifierDoc extends Doc {
     if (!this.pkg) {
       return builder;
     }
+    // builder.append(
+    // `[![npm](https://img.shields.io/npm/dt/@unibeautify/beautifier-php-cs-fixer.svg?style=for-the-badge)](https://www.npmjs.com/package/@unibeautify/beautifier-php-cs-fixer)`,
```
Comment (Glavin001) [inline scripts/generate-docs/BeautifierDoc.ts]: Remove old code.

File: scripts/generate-docs/BeautifierDoc.ts [hunk 4]
```diff
@@ -81,43 +123,104 @@ export default class BeautifierDoc extends Doc {
...
-            },
-            null,
-            2
-          ),
-          "json"
-        );
-      });
-    });
+
+    // const { dependencies } = this; if (dependencies.length === 0) {   return
```
Comment (Glavin001) [inline scripts/generate-docs/BeautifierDoc.ts]: Remove old code.

File: scripts/generate-docs/ExecutableDoc.ts [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+    builder.details("Show me how to open the Command Prompt.", builder => {
+      builder.append(
+        "\n<p><iframe width='560' height='315' src='https://www.youtube.com/embed/MBBWV" +
+          "gE0ewk' frameborder='0' allow='autoplay; encrypted-media' allowfullscreen></if" +
+          "rame></p>\n"
+      );
+    });
+    builder.append("");
+
+    // builder.append("It looks like the following for Windows 7:"); builder.append(
```
Comment (Glavin001) [inline scripts/generate-docs/ExecutableDoc.ts]: Remove old code.

File: scripts/generate-docs/ExecutableDoc.ts [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+        dependency.program
+      )}\` with your specific executable path value.`
+    );
+    return builder;
+  }
+
+  private appendMacSection(builder: MarkdownBuilder): MarkdownBuilder {
+    builder.header("macOS", 3);
+    const dependency = this.executable;
+    // builder.append("![mac-terminal](/img/mac/mac-terminal.png)");
```
Comment (Glavin001) [inline scripts/generate-docs/ExecutableDoc.ts]: Remove old code.

File: scripts/generate-docs/ExecutableDoc.ts [hunk 7]
```diff
@@ -0,0 +1,375 @@
...
+
+    return builder;
+  }
+
+  private appendLinuxSection(builder: MarkdownBuilder): MarkdownBuilder {
+    builder.header("Linux", 3);
+    const dependency = this.executable;
+    builder.append("Open the Terminal application.");
+    builder.details("Show me how to open the Terminal.", builder => {
+      // builder.append( "![linux-terminal](/img/linux/linux-terminal-on-ubuntu.png)",
```
Comment (Glavin001) [inline scripts/generate-docs/ExecutableDoc.ts]: Remove old code.

File: scripts/generate-docs/MarkdownBuilder.ts [hunk 8]
```diff
@@ -33,16 +35,44 @@ export default class MarkdownBuilder {
...
+  }
   public editButton(text: string, url: string): MarkdownBuilder {
     const button = `<a class="edit-page-link button" href="${url}" target="_blank">${text}</a>`;
     this.append(`<div>${button}</div>`);
     this.append("");
     return this;
   }
+  public appendBadges(badges: Badge[]): MarkdownBuilder {
+    if (badges.length > 0) {
+      // this.append(`| ${badges.map(badge => badge.description).join(" | ")} |`);
```
Comment (Glavin001) [inline scripts/generate-docs/MarkdownBuilder.ts]: Remove old code.

File: scripts/generate-docs/MarkdownBuilder.ts [hunk 9]
```diff
@@ -33,16 +35,44 @@ export default class MarkdownBuilder {
...
+      this.append("");
+    }
+    return this;
+  }
   public static createDocLink(text: string, docId: string): string {
     return this.createLink(text, `/docs/${docId}.html`);
   }
+  public static createBadge(badge: Badge): string {
+    const { url, href, description } = badge;
+    // return this.createLink(this.createImage(url, description), href);
```
Comment (Glavin001) [inline scripts/generate-docs/MarkdownBuilder.ts]: Remove old code.

File: scripts/generate-docs/OptionsDoc.ts [hunk 10]
```diff
@@ -64,12 +65,20 @@ export default class OptionsDoc extends Doc {
...
+    builder.append(`**Config Key**: \`${this.optionKey}\`\n`);
     builder.append(`**Description**: ${this.option.description}\n`);
     if (this.option.deprecated) {
-      builder.append(`**Deprecated since version**: ${this.option.deprecated}\n`);
+      builder.append(
+        `**Deprecated since version**: ${this.option.deprecated}\n`,
+      );
     }
-    builder.append(`**Available since version**: ${this.option.since}\n`);
+    // const currentUnibeautifyVersionBadge = '<a href="https://www.npmjs.com/package/unibeautify"><img style="display: inline;" src="https://img.shields.io/npm/v/unibeautify.svg" alt="npm"></a>';
```
Comment (Glavin001) [inline scripts/generate-docs/OptionsDoc.ts]: Remove old code.

File: website/static/css/custom.css [hunk 11]
```diff
@@ -74,4 +74,24 @@ a {
...
 .select select:disabled~.select__arrow {
   border-top-color: #ccc;
+}
+
+details {
+  margin-top: 1rem;
+  /* margin-left: 1rem; */
+}
+
+/* details[open] {
```
Comment (Glavin001) [inline website/static/css/custom.css]: Remove old code.

File: website/static/js/selectList.js [hunk 12]
```diff
@@ -1,24 +1,31 @@
...
+    );
     updateCodeExample(languageName);
-  }
-}
+  };
+};
 
 function loadSelected() {
-  const language = getLanguageFromUrl();
+  const language = getLanguageFromUrlParam(); // || getLanguageFromHash();
```
Comment (Glavin001) [inline website/static/js/selectList.js]: Remove old code.

File: website/static/js/selectList.js [hunk 13]
```diff
@@ -1,24 +1,31 @@
...
   language && updateCodeExample(language);
 }
 
 window.onpopstate = function onPageChange(event) {
   const state = event.state || {};
-  const { language: selectedLanguage } = state;
+  const {
+    language: selectedLanguage, //= getLanguageFromHash()
```
Comment (Glavin001) [inline website/static/js/selectList.js]: Remove old code.

File: docs/config-file.md (hunk 2 above)
Comment (szeck87) [reply docs/config-file.md]: I think it's fine what you have here.  `options-for-language` and `options-for-beautifier`.

**Pull request merged**